File: scripts/exa_research.ts
#!/usr/bin/env -S node --loader ts-node/esm
/**
 * Exa Research Helper (Stub)
 *
 * Purpose:
 * - Demonstrate how to call Exa's API using an API key from env.
 * - Provide commented structure for future expansion.
 *
 * TODO:
 * - Confirm endpoint paths and request schema via Rube MCP.
 */

import fetch from 'node-fetch';

async function main(): Promise<void> {
  const apiKey = process.env.EXA_API_KEY;
  if (!apiKey) {
    console.error('Missing EXA_API_KEY in environment');
    process.exit(2);
  }

  const endpoint = process.env.EXA_ENDPOINT || 'https://api.exa.ai/v1/search'; // TODO confirm
  const body = {
    query: 'example search query',
    size: 5,
  };

  const res = await fetch(endpoint, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${apiKey}`,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  });

  if (!res.ok) {
    const text = await res.text();
    console.error('Request failed', res.status, text);
    process.exit(1);
  }

  const data = await res.json();
  console.log(JSON.stringify(data, null, 2));
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: scripts/exa_research.py
#!/usr/bin/env python3
"""
Explainer:
  Exa-backed research helper. Queries the Exa Search API to retrieve samples.
  This is a placeholder pending authoritative schema from Rube MCP / exa.ai docs.

Usage:
  python scripts/exa_research.py --query "vector databases" --limit 5

# TODO: Replace BASE_URL, endpoints, and fields with official Exa API spec.
"""
from __future__ import annotations

import argparse
import json
import os
import sys
from typing import Any, Dict

import urllib.request

BASE_URL = os.environ.get("EXA_BASE_URL", "https://api.exa.ai")  # TODO
API_KEY = os.environ.get("EXA_API_KEY", "")  # TODO: Provide via secrets


def http_post_json(url: str, payload: Dict[str, Any], headers: Dict[str, str]) -> Dict[str, Any]:
    data = json.dumps(payload).encode("utf-8")
    req = urllib.request.Request(url, data=data, headers=headers, method="POST")
    with urllib.request.urlopen(req) as resp:
        body = resp.read().decode("utf-8")
        return json.loads(body)


def main() -> int:
    parser = argparse.ArgumentParser()
    parser.add_argument("--query", required=True)
    parser.add_argument("--limit", type=int, default=5)
    args = parser.parse_args()

    if not API_KEY:
        print("ERROR: EXA_API_KEY not set", file=sys.stderr)
        return 2

    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {API_KEY}",  # TODO: Confirm auth scheme
    }
    payload = {"query": args.query, "num_results": args.limit}  # TODO: Confirm fields

    try:
        resp = http_post_json(f"{BASE_URL}/search", payload, headers)  # TODO: Confirm path
    except Exception as exc:
        print(f"Request failed: {exc}", file=sys.stderr)
        return 1

    print(json.dumps(resp, indent=2))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

